File: python/src/multiagent_boilerplate/agents/lead_agent.py
from __future__ import annotations

from ..harness import Harness
from ..providers.types import ChatModel
from ..tools.types import ToolRuntime
from ..tracing.tracer import Tracer
from .run_agent import RunAgentParams, run_agent
from .types import AgentResult


async def run_lead_agent(
    query: str,
    model: ChatModel,
    harness: Harness,
    runtime: ToolRuntime,
    tracer: Tracer,
    run_id: str,
    parent_span_id: str | None = None,
) -> AgentResult:
    return await run_agent(
        RunAgentParams(
            role_name="lead",
            user_prompt=query,
            model=model,
            harness=harness,
            runtime=runtime,
            tracer=tracer,
            task_id=run_id,
            delegation_depth=0,
            parent_span_id=parent_span_id,
        )
    )

File: typescript/src/agents/leadAgent.ts
import type { ChatModel } from "../providers/types.js";
import type { Harness, RunBudget } from "../harness/index.js";
import type { ToolRuntime } from "../tools/types.js";
import type { Tracer } from "../tracing/tracer.js";
import { runAgent } from "./runAgent.js";
import type { AgentResult } from "./types.js";

export interface RunLeadAgentParams {
  query: string;
  model: ChatModel;
  harness: Harness;
  runtime: ToolRuntime;
  tracer: Tracer;
  runId: string;
  parentSpanId?: string | null;
  runBudget?: RunBudget;
}

export async function runLeadAgent(params: RunLeadAgentParams): Promise<AgentResult> {
  return runAgent({
    roleName: "lead",
    userPrompt: params.query,
    model: params.model,
    harness: params.harness,
    runtime: params.runtime,
    tracer: params.tracer,
    taskId: params.runId,
    delegationDepth: 0,
    parentSpanId: params.parentSpanId ?? null,
    ...(params.runBudget ? { runBudget: params.runBudget } : {}),
  });
}
